demos/python_demos/sound_classification_demo/sound_classification_demo.py

```diff
@@ -0,0 +1,226 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from argparse import ArgumentParser, SUPPRESS
+import logging
+import sys
+import time
+import wave
+
+import numpy as np
+from openvino.inference_engine import IECore
+
+
+def type_overlap(arg):
+    if arg.endswith('%'):
+        res = float(arg[:-1]) / 100
+    else:
+        res = int(arg)
+    return res
+
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    args = parser.add_argument_group('Options')
+    args.add_argument('-h', '--help', action='help', default=SUPPRESS, help='Show this help message and exit.')
+
+    args.add_argument('-i', '--input', type=str, required=True,
+                      help="Required. Input to process")
+    args.add_argument('-m', "--model", type=str, required=True,
+                      help="Required. Path to an .xml file with a trained model.")
+    args.add_argument("-l", "--cpu_extension", type=str, default=None,
+                      help="Optional. Required for CPU custom layers. Absolute path to a shared library with "
+                           "the kernels implementations.")
+    args.add_argument("-d", "--device", type=str, default="CPU",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL or MYRIAD is"
+                           " acceptable. The sample will look for a suitable plugin for device specified. "
+                           "Default value is CPU")
+    args.add_argument('--labels', type=str, default=None,
+                      help="Optional. Labels mapping file")
+    args.add_argument('-sr', '--samplerate', type=int,
+                      help="Optional. Set sample rate for audio input")
+    args.add_argument('-ol', '--overlap', type=type_overlap, default=0,
+                      help='Optional. Set the overlapping between audio clip in samples or percent')
+
+    return parser.parse_args()
+
+
+class AudioSource:
+    def __init__(self, source, channels=2, samplerate=None):
+        self.samplerate = samplerate
+        samplerate, audio = read_wav(source, as_float=True)
+        audio = audio.T
+        if audio.shape[0] != channels:
+            raise RuntimeError("Audio has unsupported number of channels - {} (expected {})"
+                               .format(audio.shape[0], channels))
+        if self.samplerate:
+            if self.samplerate != samplerate:
+                audio = resample(audio, samplerate, self.samplerate)
+        else:
+            self.samplerate = samplerate
+
+        self.audio = audio
+
+    def duration(self):
+        return self.audio.shape[1] / self.samplerate
+
+    def chunks(self, size, hop=None, num_chunks=1):
+        def get_clip(pos, size):
+            if pos > self.audio.shape[1]:
```

Comment:
Isn't this an impossible condition? `get_clip` is only called from within the while loop whose condition is `pos < self.audio.shape[1]`.

Reply:
It could appear for batch size > 1. In that case will be generated blank audio chunk and it won't be printed as a result.

Reply:
Right. I missed that.